cmd/rekor_monitor/main.go

```diff
@@ -76,12 +69,16 @@ func main() {
 		if err := yaml.Unmarshal([]byte(configString), &config); err != nil {
 			log.Fatalf("error parsing identities: %v", err)
 		}
+
+		configProvided = true
```

Comment:
Do we need this? An uninitalized `config` should be sufficient for the checks below like `config.EndIndex == nil`

Reply:
I believe that in the event that an uninitialized config is currently passed in, the workflow will still attempt to set `StartIndex` and `EndIndex` using the latest checkpoint and log checkpoint file and then attempt to run the identity monitor using the uninitialized config's values resulting in an error- if we would prefer to have checking for the config's indices stop the identity monitor from running I can refactor that in another PR after this one

Reply:
> an uninitialized config will have the start and end indices set and then the identity monitor will run with nil arguments

I thought the defaults for start and end were [nil](https://github.com/sigstore/rekor-monitor/blob/114befbd5a9ab2639b80d592f7c777791495c192/pkg/notifications/notifications.go#L69-L70). So if the config is not initialized, then the default values of `nil` will be read in the code you linked, and then the monitor will initialize them with defaults.

Reply:
Yes - I guess what I'm thinking is that in the event that a config is not initialized, there are also no identities provided to monitor; is the identity monitor portion of the workflow then still necessary to run?

Reply:
Ah I see your question. What about if we gated calling `IdentitySearch` on if `MonitoredValues` is set? That was like the original implementation, where it would only run if a user configured a value to monitor for.